x-pack/plugins/ml/server/models/job_validation/messages.js

```diff
@@ -0,0 +1,410 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import { i18n } from '@kbn/i18n';
+
+let messages;
```

Comment:
Can you explain why you chose this approach? Is it to render the messages only on demand once?
Let me know what you think of this alternative:

```javascript
export const messages = {
  field_not_aggregatable: {
    status: 'ERROR',
    text: i18n.translate('xpack.ml.models.jobValidation.messages.fieldNotAggregatableMessage', {
      defaultMessage: 'Detector field {fieldName} is not an aggregatable field.',
        values: {
          fieldName: '"{{fieldName}}"',
        },
      }),
      url: 'https://www.elastic.co/guide/en/elastic-stack-overview/{{version}}/ml-configuring-aggregation.html'
    },
   ...
}
```

Reply:
or even `export const messages = getMessages();`
if the point is to have the translations inside a function

Reply:
Thanks for clarifying!